### ISSUE TITLE
Training TAPIR PyTorch version script?

### ISSUE THREAD

**Issue by riponazad** [google-deepmind/tapnet#90]
Is there any script for training TAPIR pytorch version?

**Comment by cdoersch** [comment]
Internally we train using JAX.  Maintaining a second training codebase would be non-trivial work for us and is beyond the capacity of our team.  Sorry about that.

**Comment by riponazad** [comment]
Thanks for your response. Yes, I got that. I tried to train with the existing PyTorch TAPIR code but failed to optimize due to the non-integration structure. Seems need to spend more time now. Anyway thanks for the nice work.

**Comment by justachetan** [comment]
Hi @riponazad could you please explain what you mean by the non-integration structure? Thanks!

**Comment by riponazad** [comment]
Hi @justachetan, the provided pytorch script doesn't track the computation during the forward pass. Hence, autograd can't compute gradients and the optimizer doesn't update the parameters of the model. That is my understanding and I didn't dig anymore. I may be wrong or describing wrongly. Please let me know if you are abale to train it.

**Comment by justachetan** [comment]
Thanks! Yeah I realized that too. But I guess if you remove the calls to `.detach()` in the code that should be fixed, right @cdoersch?

**Comment by riponazad** [comment]
@justachetan Thanks a lot :). That's it. Saved a lot of time from digging. Now it's being optimized.

**Comment by shivanimall** [comment]
Hi @justachetan @riponazad in case you have written out a training script in torch, I wanted to see if it is possible to make it available? (I wanted to compare with what I have) thank you!

**Comment by yangyi02** [comment]
Hi, may I also ask how is the work with training script in pytorch? We would sincerely welcome your implementation of tapir training in pytorch. Thanks.

**Comment by riponazad** [comment]
> Hi, may I also ask how is the work with training script in pytorch? We would sincerely welcome your implementation of tapir training in pytorch. Thanks.

Hello,  
I fine-tuned the model using a wrapper class specific to my project but didn't integrate it directly into this codebase. If anyone is interested in training or fine-tuning code, I have just pushed it in [EchoTracker](https://github.com/riponazad/echotracker?tab=readme-ov-file#trainevaluation).  

And, adapting the code for this repository will take me some time, but if desired, I'd be happy to work on it later. 😊

**Comment by yangyi02** [comment]
Oh, thanks for sharing, @riponazad This is interesting, good work!

**Comment by yangyi02** [comment]
@riponazad We are currently not planning to release a pytorch traning code. Do you mind we link your repo as a reference for pytorch implementation of TAPIR?

**Comment by riponazad** [comment]
@yangyi02 Hello, sure, no problem at all. That will be even better for me. Thanks.
